File: frontend/src/services/popupService.ts
import client from '@/lib/api/client';
import { Popup } from '@/types/banner';

export const popupService = {
  getPopups: async (params: { page?: number; size?: number; searchWrd?: string }, config?: any) => {
    return client.get<any>('/popups', { ...config, params });
  },

  getActivePopups: async (config?: any) => {
    return client.get<any>('/popups/active', config);
  },

  getPopup: async (popupId: string, config?: any) => {
    return client.get<any>(`/popups/${popupId}`, config);
  },

  createPopup: async (data: Popup, config?: any) => {
    return client.post('/popups', data, config);
  },

  updatePopup: async (popupId: string, data: Popup, config?: any) => {
    return client.put(`/popups/${popupId}`, data, config);
  },

  deletePopup: async (popupId: string, config?: any) => {
    return client.delete(`/popups/${popupId}`, config);
  }
};

File: module-system-admin/src/main/java/com/company/project/service/system/content/popup/PopupService.java
package com.company.project.service.system.content.popup;

import com.company.project.service.system.content.popup.dto.PopupDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.List;

public interface PopupService {
    Page<PopupDto> getPopupList(String keyword, Pageable pageable);

    List<PopupDto> getActivePopups();

    PopupDto getPopup(String popupId);

    String createPopup(String userId, PopupDto dto);

    void updatePopup(String popupId, String userId, PopupDto dto);

    void deletePopup(String popupId);

    List<String> getPopupWhiteList();
}
